### ISSUE TITLE
FR: Update the architecture diagram for SLS & CDK

### ISSUE THREAD

**Issue by mikestopcontinues** [serverless-nextjs/serverless-next.js#1414]
**Is your feature request related to a problem? Please describe.**
As a potential user and potential contributor, I'd like to see the state of the SLS and CDK implementations, as well as planned improvements. This would help me decide the right course of action for my app, and also help me to see where my contributions can be applied.

**Describe the solution you'd like**
I'd like to see updated architecture diagrams.

**Describe alternatives you've considered**
Much more thorough docs. But I see the diagrams as easier to produce, easier to update, and easier to read.

**Additional context**
As a CDK user, I'd really prefer to stick with the CDK for this app. But it seems the SLS implementation has progressed far further. I really want to be clear how much further. Even the feature checklist completely ignores the CDK. It may be easy to reach parity. Or maybe not.

The question also makes me wonder if CDK can be used under the hood of SLS, thereby keeping everything in sync. I realize CDK deploys are a minute or two slower than SLS, but that's a small price to pay for the difference in code complexity between using the CDK and using the SDK to deploy infra.

**Comment by dphang** [comment]
@mikestopcontinues yeah, that is a great idea. I hadn't prioritized it yet since it's mainly me working on it, but it would be good to update the docs.

FYI, the assets/code deployed should be the same, just that CDK is specifying how to deploy - so I suppose it may be less configurable (like adding tags, etc.). But if there is a missing feature then it would be good to document and try to bring them into parity (although in the future it may be better to maintain a single solution).

Let me see if I can find some time in the next couple weeks.

**Comment by mikestopcontinues** [comment]
Hey @dphang, thanks for getting back to me. I'm a few months from launching my biz, so right now I'm completely swamped, but in the future, I would like to help out on the project. I think it's important to have a strong open source competitor to Vercel for hosting Next apps.

For now, I'm actually unable to use it because I have a multi-domain use-case. With AWS quotas, I need a single CF distro for each domain, as users sign up. That means reusing those policies across Next apps is essential. But it also means for pushing out rapid code updates, it makes more sense to use API gateway than @edge. (@edge can't point to lambda $latest, so every one of those distros would need to be redeployed every time I fixed a bug.)

I've seen you discuss (1) using CDK under the hood and (2) supporting API gateway. I hope I can contribute to that going forward. Given the hurtles I've come across, I think it also makes sense to break up the CDK construct (roughly) into: 1. Policies 2. Lambdas 3. Distribution/DNS 4. API Gateway/DNS (if it happens). This would make the codebase maximally flexible and quota-efficient. Let the pieces be wired together in different ways, even across multiple AWS sub-accounts. Right now, the divisions fall into place rather nicely aside from the BUILD_ID bucket, which causes a circular stack reference how it's implemented now. I think deploying a lambda with an S3 trigger would mitigate the problem.

My current solution (proxy domain.com => siteId.host.com) is far from ideal, so I'll definitely be circling back, but in the meantime, I just wanted to get my use-case out there.

Anyway, thanks again.
